File: Arhitecture/AsmParser/src/SymTableParser.java
public class SymTableParser {
    public static String getBind(final int code) {
        return switch (code) {
            case 0 -> "LOCAL";
            case 1 -> "GLOBAL";
            case 2 -> "WEAK";
            case 10 -> "LOOS";
            case 12 -> "HIOS";
            case 13 -> "LOPROC";
            case 15 -> "HIPROC";
            default -> throw new IllegalArgumentException(String.format("Unsupported code %s for bind", code));
        };
    }

    public static String getType(final int code) {
        return switch (code) {
            case 0 -> "NOTYPE";
            case 1 -> "OBJECT";
            case 2 -> "FUNC";
            case 3 -> "SECTION";
            case 4 -> "FILE";
            case 5 -> "COMMON";
            case 6 -> "TLS";
            case 10 -> "LOOS";
            case 12 -> "HIOS";
            case 13 -> "LOPROC";
            case 15 -> "HIPROC";
            default -> throw new IllegalArgumentException(String.format("Unsupported code %s for type", code));
        };
    }

    public static String getVisibility(final int code) {
        return switch (code) {
            case 0 -> "DEFAULT";
            case 1 -> "INTERNAL";
            case 2 -> "HIDDEN";
            case 3 -> "PROTECTED";
            default -> throw new IllegalArgumentException(String.format("Unsupported code %s for visibility", code));
        };
    }

    public static String getId(final int code) {
        return switch (code) {
            case 0 -> "UNDEF";
            case 0xff00 -> "LOPROC";
            case 0xff01 -> "AFTER";
            case 0xff02 -> "AMD64_LCOMMON";
            case 0xff1f -> "HIPROC";
            case 0xff20 -> "LOOS";
            case 0xff3f -> "HIOS";
            case 0xfff1 -> "ABS";
            case 0xfff2 -> "COMMON";
            case 0xffff -> "XINDEX";
            default -> String.valueOf(code);
        };
    }
}

File: Arhitecture/Lab3/parser/SymtableParser.cpp
//
// Created by mike on 12/23/22.
//

#include "SymtableParser.h"
#include <stdexcept>


std::string utility::SymtableParser::GetBind(int code) {
    switch (code) {
        case 0:
            return "LOCAL";
        case 1:
            return "GLOBAL";
        case 2:
            return "WEAK";
        case 10:
            return "LOOS";
        case 12:
            return "HIOS";
        case 13:
            return "LOPROC";
        case 15:
            return "HIPROC";
    }
    throw std::invalid_argument(&"Unsupported symtable bind code = "[code]);
}

std::string utility::SymtableParser::GetType(int code) {
    switch (code) {
        case 0:
            return "NOTYPE";
        case 1:
            return "OBJECT";
        case 2:
            return "FUNC";
        case 3:
            return "SECTION";
        case 4:
            return "FILE";
        case 5:
            return "COMMON";
        case 6:
            return "TLS";
        case 10:
            return "LOOS";
        case 12:
            return "HIOS";
        case 13:
            return "LOPROC";
        case 15:
            return "HIPROC";
    }
    throw std::invalid_argument(&"Unsupported symtable type code = "[code]);
}

std::string utility::SymtableParser::GetVisibility(int code) {
    switch (code) {
        case 0:
            return "DEFAULT";
        case 1:
            return "INTERNAL";
        case 2:
            return "HIDDEN";
        case 3:
            return "PROTECTED";
    }
    throw std::invalid_argument(&"Unsupported symtable visibility code = "[code]);
}

std::string utility::SymtableParser::GetId(int code) {
    switch (code) {
        case 0:
            return "UNDEF";
        case 0xff00:
            return "LOPROC";
        case 0xff01:
            return "AFTER";
        case 0xff02:
            return "AMD64_LCOMMON";
        case 0xff1f:
            return "HIPROC";
        case 0xff20:
            return "LOOS";
        case 0xff3f:
            return "HIOS";
        case 0xfff1:
            return "ABS";
        case 0xfff2:
            return "COMMON";
        case 0xffff:
            return "XINDEX";
    }
    return std::to_string(code);
}
